### PR TITLE
Preserve Downloader.paused state on restart

### LINKED ISSUE #1062: Feature request: preserve pause state on shutdown/restart
Just update to 2.3.1~rc2 on my Ubuntu 17.04 server. Before the update SABnzbd was paused. After the update SABnzbd automatically restarted and began downloading again.

I have a monthly data cap, if I didn't have a telegram bot I wouldn't have known that I was about to blast through it. 

Not sure if this is a bug or a feature request.

### PR DESCRIPTION
**Pull request by puzzledsab**
Fixes #1062.

A drop down box with three options; Resume, Paused and Preserve state would be prettier but it would still need two variables.

### CODE REVIEW

File: sabnzbd/cfg.py [hunk 1]
```diff
@@ -260,6 +260,7 @@ def validate_script(value):
 ampm = OptionBool("misc", "ampm", False)
 replace_illegal = OptionBool("misc", "replace_illegal", True)
 start_paused = OptionBool("misc", "start_paused", False)
+preserve_state = OptionBool("misc", "preserve_state", False)
```
Comment (Safihre) [inline sabnzbd/cfg.py:263]: Let's make it a bit more clear what it means: `preserve_paused_state`

File: sabnzbd/downloader.py [hunk 2]
```diff
@@ -372,12 +372,17 @@ def remove_socket(self, nw: NewsWrapper):
     @NzbQueueLocker
     def set_paused_state(self, state: bool):
         """Set downloader to specified paused state"""
-        self.paused = state
+        if self.paused != state:
```
Comment (Safihre) [inline sabnzbd/downloader.py:375]: What happens here? Why is this needed?
This code needs come comments.

File: sabnzbd/downloader.py [hunk 3]
```diff
@@ -372,12 +372,17 @@ def remove_socket(self, nw: NewsWrapper):
...
+            if cfg.preserve_state():
+                cfg.start_paused.set(int(state))
+            self.paused = state
 
     @NzbQueueLocker
     def resume(self):
         # Do not notify when SABnzbd is still starting
         if self.paused and sabnzbd.WEB_DIR:
+            if cfg.preserve_state():
+                cfg.start_paused.set(0)
```
Comment (Safihre) [inline sabnzbd/downloader.py:385]: `start_paused` is an `OptionBool` so use `.set(False)` and `.set(True)`.
And let's put it below the `sabnzbd.notifier.send_notification`

File: sabnzbd/downloader.py [hunk 4]
```diff
@@ -386,6 +391,8 @@ def resume(self):
     def pause(self):
         """Pause the downloader, optionally saving admin"""
         if not self.paused:
+            if cfg.preserve_state():
+                cfg.start_paused.set(1)
```
Comment (Safihre) [inline sabnzbd/downloader.py:395]: Same comments as the `resume`.

**Review by Safihre — CHANGES_REQUESTED** [review]
Just so small changes requested.

File: sabnzbd/downloader.py (hunk 2 above)
Comment (puzzledsab) [reply sabnzbd/downloader.py:375]: It's not strictly needed but it avoids doing the check and update if there is no change. It's also not necessary to set paused if it's unchanged so I moved it to the same level.

**Review by Safihre — APPROVED** [review]

**Comment by Safihre** [comment]
Could you also add it to the documentation for 3.5?
